src/components/TradeWidget/index.tsx

```diff
@@ -56,6 +56,23 @@ const SubmitButton = styled.button`
   }
 `
 
+// This is sort of a duplication from the validation done on TokenRow.
+// Still, required because the initial value comes from the URL params,
+//   and we have no control over what the user inserts there.
+// Thus, we make it a nicely formated value, or 0 if we can't
+// And only on first render. Subsequent validation is taken care by TokenRow.
+function sanitizeInputAmount(value: string, precision: number): string {
```

Comment:
Would it make sense to be moved to our utils?

Reply:
Did a small refactoring, let me know if you still think this should be in utils.